nimble/controller/src/ble_ll_conn_hci.c

```diff
@@ -1260,13 +1260,12 @@ ble_ll_conn_hci_set_data_len(uint8_t *cmdbuf, uint8_t *rspbuf, uint8_t *rsplen)
             rc = BLE_ERR_SUCCESS;
         }
 
-        /* XXX: should I check against max supported? I think so */
+        if (connsm->max_tx_time != txtime || connsm->max_tx_octets != txoctets) {
```

Comment:
this should not be executed when command returns an invalid parameters error

Reply:
ah right.